### PR TITLE
React: Refactor /api/groups hookup to use react-query

### PR DESCRIPTION
**Pull request by frewmack**
These custom hooks each correspond to the 'groups' endpoints, and help for managing fetching behaviour in one place.

### CODE REVIEW

**Review by kevinlul — COMMENTED** [review]
Is naming the hook after the HTTP method the best or a verb for what the action actually is (create, update)?

**Comment by frewmack** [comment]
> Is naming the hook after the HTTP method the best or a verb for what the action actually is (create, update)?

I did have an idea to include an 'alias' hook called `useGroupsEndpoint` that provides `.post(), .patch(), .delete()` which just call the `.mutate()` functions for their respective mutations.

Other than that, I don't know what the standard is. Even using a verb is inaccurate since it returns a mutation object that you then use to 'mutate' the query data by interacting with the backend.

**Comment by kevinlul** [comment]
Okay, I guess as-is is fine.

File: react/src/Groups/index.tsx [hunk 1]
```diff
@@ -31,17 +31,16 @@ const useStyles = makeStyles((theme: Theme) => ({
 
 export default function Groups() {
     const classes = useStyles();
-    const [groups, setGroups] = useState<Group[]>([]);
-    const [openNewGroup, setOpenNewGroup] = useState<boolean>(false);
+    const groups = useGroups();
+    const groupPatch = useGroupsPatch();
+    const groupDelete = useGroupsDelete();
+    const [openNewGroup, setOpenNewGroup] = useState(false);
```
Comment (kevinlul) [inline react/src/Groups/index.tsx:37]: It seems like on the rerender now when we add or remove a group, `/api/groups` and each group are re-fetched now and I guess this makes sense due to cache invalidation but I don't think this was the behaviour before?

Comment (frewmack) [reply react/src/Groups/index.tsx:37]: Yeah this wasn't the behaviour before. I'm looking into it.

**Review by kevinlul — APPROVED** [review]
LGTM!

**Pull request merged**